app/services/spree_shopify_importer/data_savers/zones/zone_base.rb

```diff
@@ -0,0 +1,74 @@
+module SpreeShopifyImporter
+  module DataSavers
+    module Zones
+      class ZoneBase < BaseDataSaver
+        delegate :attributes, to: :parser
+
+        private
+
+        def assign_spree_zone_to_data_feed
+          @shopify_data_feed.update!(spree_object: @spree_zone)
+        end
+
+        def parser
+          @parser ||= SpreeShopifyImporter::DataParsers::Zones::BaseData.new(shopify_shipping_zone)
+        end
+
+        def shopify_shipping_zone
+          @shopify_shipping_zone ||= ShopifyAPI::ShippingZone.new(data_feed)
+        end
+
+        def create_spree_zone_members
+          if attributes[:kind] == 'country'
```

Comment:
since we are using the same strings across all the codebase I would like to suggest to create a file like `shipping_zones/constants.rb` and we can put there as minimum 2 constants for now.

```
  COUNTRY = 'country'
  STATE = 'state'
```

and then in any place `ShippingZones::Constants::STATE`

Reply:
also may be
```
  attributes[:kind] == ShippingZones::Constants::STATE ? find_countries_members : find_states_members
```

ternary operator looks good here